app/connectors/connector-utils/src/main/java/io/syndesis/connector/utils/KeyStoreHelper.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.connector.utils;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.CertificateException;
+import java.util.stream.Collectors;
+
+/**
+ * Helper class to create KeyStores from Certificates.
+ *
+ * @author dhirajsb
+ */
+public class KeyStoreHelper {
+
+    private static final SecureRandom SECURE_RANDOM = new SecureRandom();
```

Comment:
Could use [SecureRandom.getInstanceStrong()](https://docs.oracle.com/javase/8/docs/api/java/security/SecureRandom.html#getInstanceStrong--), though if it's used for generating passwords, [ThreadLocalRandom.current()](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadLocalRandom.html#current--) ought to be enough random.

Reply:
I used to think that `getInstanceStrong()` was preferable too, until I read [this](https://tersesystems.com/blog/2015/12/17/the-right-way-to-use-securerandom/).

Reply:
BTW, we also do [this](https://github.com/fabric8io-images/s2i/blob/39f99bf1755809069576233e914f3bc0eb7fd8d3/java/images/rhel/Dockerfile#L29).